### DISCUSSION TITLE
Custom HTTP response codes

### DISCUSSION THREAD

**Discussion by mael-queau** [phero-hq/phero#89] [Q&A] [answered]
Hi, I've only just discovered Phero, and it looks like an astonishing project, but I had one preliminary question about its capabilities. Would it be possible to specify custom error codes, for example in the case of 404 errors and such, or is it all done through the use of custom error types?

I'd like to be able to use the API with software written in languages other than TypeScript, for instance, and having the ability to specify the proper HTTP response codes would go a long way.

Thanks in advance!

**Comment by JasperH8g** [comment, marked as answer]
Excellent question! Optimising phero-server to be consumed by something different than phero-client (indeed by allowing to set response-codes and things like headers) is something that we've been discussing internally: https://github.com/phero-hq/phero/issues/53. So, to answer your question: We don't support this at the moment, but it's a matter of time 😉

**Reply by JasperH8g** [reply]
Oh, by the way: We do use some status-codes (200 for when everything's fine, 400 when there're validation-errors in the request or response and 500 if something else went wrong), but those are not customisable for a function by the developer at the moment.

**Reply by mael-queau** [reply]
I see, thanks a lot for the (very) quick response :)

**Reply by mael-queau** [reply]
There are numerous things that would be very interesting to implement the more I think about it, for example a more streamlined way to implement middlewares for authorization purposes, for instance.

So far, I don't believe there is a way other than manually calling a middleware function from within every function that requires it, creating some duplicate parameters in the case of an API Key being used for example (albeit the specific question of api keys would be made easier with custom request headers).
